Fix overwriting for vtkjs files

DavidBerger98 opened the pull request:
When exporting to vtkjs, files in the data folder ending with .gz would not be overwritten

laurennlam commented on dicomexporter/exporter.py:
```diff
@@ -25,12 +27,28 @@ def convertDICOMVolumeToVTKFile(
+            )
+            return False, None
+        else:
+            if file_extension == '.vti':
```
.vti is already defined in ALLOWED_EXTENSIONS. That should be better to only define once the allowed extensions. That would avoid error when we want to update it. Maybe using a dictionary or something like this.
Kind of `if file_extension == ALLOWED_EXTENSION.vti`. (Not sure about the syntax)

finetjul replied:
I kind of agree with you.

To make things robust, please consider that some file formats may have multiple file_extensions (e.g. mha, mhd)

DavidBerger98 replied:
Good point! I implemented it as a separate commit

laurennlam commented on dicomexporter/exporter.py:
```diff
@@ -25,12 +27,28 @@ def convertDICOMVolumeToVTKFile(
 
+    # Test output_file_path #
+    if os.path.exists(output_file_path):
+        if not overwrite:
```
It's just a remark, you don't have to change it if it's good for you, but when there is only if/else, I think it's more readable to not using "not" in the if. But it's just my point of view. Not necesarily the best ;) You can leave it as it.

finetjul replied:
@laurennlam we came up with the "NIP" (i.e. not important) prefix, to these kind of "preference"/"not important" remarks. you avoid therefore the long "disclaimer speech" :-)

And for what it worthes (i.e. NIP), I personally like to handle errors first rather than later :-)

DavidBerger98 replied:
Thank you for the remark. Actually the else branch isn't even necessary since we return in the if-case, so I removed it. Makes it more readable that way.

Commit:
Check if extension is allowed before conversion

Implement ALLOWED_EXTENSIONS enum to make checking for extensions more robust

Commit:
Fix overwriting for vtkjs files

When exporting to vtkjs, files in the data folder ending with .gz would not be overwritten

laurennlam commented on dicomexporter/exporter.py:
```diff
@@ -13,6 +13,31 @@
+    vtkjs = 'vtkjs'
+
+
+def splitext(path):
```
I would prefer a name like 'extractExtensionsFromFilePath', something more readable

laurennlam commented on dicomexporter/exporter.py:
```diff
@@ -13,6 +13,31 @@
+def splitext(path):
+    """
+    Given a path, return it without extensions and a list of all its
+    extensions (without a leading dot '.').
```
Why a given path would have several extensions ?

DavidBerger98 replied:
Refering to what @finetjul wrote:

> To make things robust, please consider that some file formats may have multiple file_extensions (e.g. mha, mhd)

laurennlam commented on dicomexporter/exporter.py:
```diff
@@ -13,6 +13,31 @@
+        os.path.dirname(path), os.path.basename(path).split(os.extsep)[0]
+    )
+    extensions = os.path.basename(path).split(os.extsep)[1:]
+    if len(extensions) == 1:
```
I'm not sure that you need to manage this case. I think that it would be simplier to always return a tuple of string/array.
In the function description, you can just define the output as a tuple of string and array.
The array contains the extracted extensions in the order of there are defined.
It's up to the user to manage the extensions as he wants.

DavidBerger98 replied:
Agreed!
Since I am unsure what it will look like to handle files with multiple extensions, I take the last extension. Should be fine for now until we actually need to handle multiple extensions. What do you think?

laurennlam replied:
I think that it would be better to have string/array.
I could add some test to check this method with all the cases you think about.
Then we'll update it if users are not familiar with the current working.
@finetjul , WDYT ?

finetjul replied:
Let's keep it simple if possible for now (no funding for tests as of now)

DavidBerger98 replied:
@laurennlam I applied the changes as you wanted: extractExtensionsFromFilePath now always returns a tuple containing a string (the path without extensions) and a list (the extensions).
And I just take the last element of the returned extensions list and treat it as the actual extension for now.

laurennlam commented on dicomexporter/exporter.py:
```diff
@@ -25,12 +48,29 @@ def convertDICOMVolumeToVTKFile(
-        print('Output file already exist', output_file_path, 
-            '\nIf you want to overwrite the file add the \'--overwrite\' flag')
+    _, file_extensions = extractExtensionsFromFilePath(output_file_path)
+    file_extension = file_extensions[-1] # only handling single file_extensions for now
```
What's happening if file_extensions is empty ?

DavidBerger98 replied:
Missed that one. Should be fixed now!

laurennlam commented:
LGTM

finetjul commented on dicomexporter/exporter.py:
```diff
@@ -25,12 +48,30 @@ def convertDICOMVolumeToVTKFile(
-            '\nIf you want to overwrite the file add the \'--overwrite\' flag')
+    _, file_extensions = extractExtensionsFromFilePath(output_file_path)
+    # only handling single file_extensions for now
+    file_extension = file_extensions[-1] if len(file_extensions) > 0 else ''
```
NIP: could have be in a convenient `extractLastExtensionFromFilePath`
